### ISSUE TITLE
False Positive: statement seems to be disconnected from its parent foreach

### ISSUE THREAD

**Issue by klx01** [kalessil/phpinspectionsea#1698]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.6 |
| Language level | PHP 7.4  |

Current behaviour
--
```php
function foo(array $param) {
    foreach ($param as $item) {
        ['field' => $field] = $item;
        echo $field; // False Positive - [EA] This statement seems to be disconnected from its parent foreach

        $field2 = $item['field'];
        echo $field2; // no error
    }
}
```
Expected behaviour
--
No error

**Comment by jdreesen** [comment]
There's no warning if you do it like this, though:

```php
function foo(array $param) {
    foreach ($param as ['field' => $field]) {
        echo $field;
    }
}
```

**Comment by asamofal** [comment]
@jdreesen But it's not a solution when you want to destructively assign more items:
```php
foreach ($params as $item) {
    [
        'field1' => $field1,
        'field2' => $field2,
        'field3' => $field3,
    ] = $item;
        
    echo $field1; // This statement seems to be disconnected from its parent foreach.
}
```

In my case `$item` is an object... so I use `get_object_vars($item)` to destruct, it's better to keep destruction inside the body.

**Comment by asamofal** [comment]
Can anyone fix it, please? It's definitely a bug - my variable is connected to the loop just through destructuring assignment.

![image](https://user-images.githubusercontent.com/2729547/134903654-7e65e319-1b56-41a0-9eaa-d8c94d448e4c.png)

**Comment by ea-inspections-team** [comment]
Thanks for reporting @klx01! We'll look into fixing this in one of the next releases.
